### PR TITLE
Banquet for Heroes ep 16.1

### PR DESCRIPTION
**Pull request by Atemo**
* **Addressed Issue(s)**: #1305

* **Server Mode**: renewal

* **Description of Pull Request**: 

Walkthrough Conversion of Episode 16.1
* Main quest and the 2 instances Ritual of Blessing and Room of Consciousness
* Sidequests around the families, the kitchen, underground prison and old prontera
* Enchants NPCs (enchant abusive robe and co.)

Thanks to @admkakaroto and https://www.divine-pride.net/ !

TODO
- [ ] Refactor quest_db to implement hunting quest by race and level

### CODE REVIEW

File: db/re/mob_db.txt [hunk 1]
```diff
@@ -2599,20 +2599,20 @@
...
-//3445,P_ARCHER_SKELETON
-//3446,P_SKELETON
-//3447,P_SOLDIER_SKELETON
-//3448,P_AMDARAIS
-//3449,G_P_AMDARAIS
-//3450,BIJOU
-//3451,IMMOTAL_CORPS
-//3452,ZOMBIE_GUARD
+3442,FROZENWOLF,Frozenwolf,Frozen Wolf,140,80000,1,5000,5000,1,1040,1440,50,50,100,80,100,100,100,50,10,12,1,2,21,0x3885,200,1120,420,432,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,6921,2000,4640,1
+3443,TAFFY,Taffy,Taffy,145,100000,1,6000,6000,1,1445,2045,125,47,100,80,1001,100,100,50,10,12,1,2,22,0x91,200,1604,1344,2016,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,6922,2000,4639,1
```
Comment (yoonjunho72) [inline db/re/mob_db.txt]: Mob ID 3443. Value for VIT i think 1001 is too much must be 100.

**Comment by Balferian** [comment]
I found some incorrect positions NPC
Need to change/replace:
**npc/warps/cities/prontera.txt**
Change:
prontera,177,221,0	warp	prt05	2,2,prt_in,168,128
to:
prontera,177,225,0	warp	prt05	2,2,prt_in,168,128
https://youtu.be/WiCf5CSt5T4?t=194 (kRo)

**npc/kafras/kafras.txt**
Change:
prontera,162,328,3	script	Служащая Кафры::kaf_prontera	112,{
to:
prontera,152,326,4	script	Служащая Кафры::kaf_prontera	112,{

**npc/cities/prontera.txt**
Remove:
prontera,160,330,0	script	Guard#pront::prtguard	105,{

**npc/quests/quests_nameless.txt**
Change:
prontera,150,326,0	script	Wanted Notice#edq	111,{
to:
prontera,149,328,0	script	Wanted Notice#edq	111,{
https://youtu.be/WiCf5CSt5T4?t=211 (kRo)

**npc/re/guides/guides_prontera.txt**
Change:
prontera,151,330,4	duplicate(GuideProntera)	Guide#05prontera	105
to:
prontera,150,326,4	duplicate(GuideProntera)	Guide#05prontera	105

Warps (prontera) need to check official positions:
**npc/warps/cities/prontera.txt**
prontera,179,184,0	warp	prt06	2,2,prt_in,60,73
prontera,208,154,0	warp	prt09	2,2,prt_in,172,29
prontera,42,67,0	warp	prt10	2,2,prt_in,44,29

**npc/re/warps/other/s_workshop.txt**
prontera,272,108,0	warp	s_atelier0005	1,1,s_atelier,13,119

File: npc/re/quests/quests_16_1.txt [hunk 2]
```diff
@@ -3012,3 +3012,3004 @@ S_Dish:
...
+		next;
+		mes "[Bigfoot]";
+		mes "Oh, you wanted to know that. As you know, it is rare to see seafarer Walthers in other places of Rune Midgarts, except Alberta and Prontera.";
+		mes "They are good at hiding their identities other than those who are well known.";
+		next;
+		mes "[Bigfoot]";
+		mes "However, I saw the distinctive symbol of the family in ^0000cdVeins^000000.";
+		next;
+		mes "[" + strcharinfo(0) + "]";
+		mes "Veins? Who did you see there?
```
Comment (teededung) [inline npc/re/quests/quests_16_1.txt]: Missing a space in " + strcharinfo(0) + "It is nice to see you."

Comment (teededung) [inline npc/re/quests/quests_16_1.txt]: Missing a space in "...the family from^0000cdWolf Roegenburg^000000.""

File: npc/re/instances/RitualOfBlessing.txt [hunk 3]
```diff
@@ -0,0 +1,794 @@
...
+	if ('intruder_count < 3) {
+		mes "[Kronecker]";
+		mes "Peter is in a critical condition.";
+		mes "...Would you mind fighting against the intruder for us?";
+		close;
+	}
+	if (isbegin_quest(7692) == 1) {
+		mes "[Kronecker]";
+		mes "....";
+		mes ".... ^4d4dffTera Gloria^000000 is missing...";
```
Comment (teededung) [inline npc/re/instances/RitualOfBlessing.txt]: This is "Terra Gloria" not "Tera Gloria"

**Comment by Lemongrass3110** [comment]
@Atemo is #1305 fully covered with this?

**Comment by ecdarreola** [comment]
Encountered error in Magical Gear Mechanic NPC in prt_q (image below)
image

File: npc/re/quests/quests_16_1.txt [hunk 4]
```diff
@@ -0,0 +1,17204 @@
...
+		classchange( 4_EP16_GRANZ, "Kronecker G. Heine#ep16_1", bc_self );
+		classchange( 4_EP16_NIHIL, "Nyhill M. Heine#ep16_li", bc_self );
+		classchange( 1_M_JOBTESTER, "Ian Walther#ep16_lib01", bc_self );
+		sleep2 2000;
+		npctalk "You are here at last.", "Ian Walther#ep16_lib01", bc_self;
+		sleep2 3000;
+		npctalk "I have something to tell you, special guest.", "Ian Walther#ep16_lib01", bc_self;
+	}
+	if (banquet_main_quest == 24)
+		classchange( 4_EP16_GRANZ, "
```
[the remark this answers is not in the document]
Comment (Atemo) [reply npc/re/quests/quests_16_1.txt]: What do you mean? There is no Peter npc in this part.

[the remark this answers is not in the document]
Comment (teededung) [reply npc/re/quests/quests_16_1.txt]: I mean Peter Heine in the bed is not always showing. Check the line 5180 please.

**Comment by ecdarreola** [comment]
Encountered debug line
image

**Comment by Atemo** [comment]
@Balferian Prontera got reverted to its original state in 2016 and the video is from 2015
https://github.com/rathena/rathena/issues/1305

**Comment by teededung** [comment]
@Atemo Please add function check hack for NPC Dylan

**Comment by Atemo** [comment]
NPC Dylan checks the equipments and enhances directly. It doesn't need this function.

**Comment by xTakami** [comment]
Hello, how could I add level and race parameters to my quest_db?

**Comment by spinzaku** [comment]
Hello been testing this script! great work by the way. I have encountered a problem at the moment. 

I have already  talked to each royal family.. but bell keeps telling me 'You have not visited a family yet' Maybe its better to tell which family wasn't visited?

this is the char_reg_num:

150077	banquet_main_quest	0	2
150077	banquet_heine_quest	0	2
150077	banquet_nerius_quest	0	1
150077	banquet_walther_quest	0	4
150077	banquet_wigner_quest	0	5
150077	banquet_richard_quest	0	3
150077	banquet_geoborg_quest	0	4
150077	banquet_roegenburg_quest	0	4
150077	banquet_quest_cooking	0	1

**Comment by Atemo** [comment]
@spinzaku it should be fixed now thanks !

**Comment by Atemo** [comment]
The initial version has been merged. Thanks !

**Comment by Daegaladh** [comment]
Why is this on pre-re? It ends on ep13.2, so all these changes should be reverted

**Pull request merged**